packages/cel-box/src/split.ts

```diff
@@ -0,0 +1,108 @@
+import * as CEL from "@bufbuild/cel";
+import Split from "split-grid";
+import type { SplitInstance } from "split-grid";
+import YAML from "yaml";
+import { CelBox, type CelBoxOptions } from "./internal/core.js";
+
+class CelBoxSplit extends CelBox {
+  split: SplitInstance;
+
+  currentBindings?: Record<string, CEL.CelInput>;
+  currentResult?: CEL.CelResult;
+  currentResultYAML = "";
+
+  constructor(root: Element | string, options?: CelBoxOptions) {
+    super(root, options);
+
+    this.root.innerHTML = `
+      <div class="cel-box-data">
+        <div class="cel-box-content cel-box-input">
+          <h4>Bindings</h4>
+          <div class="cel-box-code" contenteditable="plaintext-only" spellcheck="false">${this.originalContent.trim()}</div>
+        </div>
+        <div class="cel-box-gutter">
+          <span class="handle"></span>
+        </div>
+        <div class="cel-box-content cel-box-output">
+          <h4>Result</h4>
+          <div class="cel-box-code"></div>
+        </div>
+      </div>
+      ${this.getExprInputHTML()}
+    `;
+
+    this.split = Split({
+      columnGutters: [
+        {
+          track: 1,
+          element: this.root.querySelector(".cel-box-gutter") as HTMLDivElement,
+        },
+      ],
+    });
+
+    this.addExprInputListener();
+    this.addErrorRenderers();
+
+    this.addListener(
+      ".cel-box-input .cel-box-code",
+      "input",
+      "CEL_BOX_BINDINGS_INPUT",
+    );
+
+    this.addTextRenderer(
+      ".cel-box-output .cel-box-code",
+      () => this.currentResultYAML,
+    );
+  }
+
+  override update(event: string, element: HTMLElement) {
+    super.update(event, element);
+
+    if (event == "CEL_BOX_BINDINGS_INPUT") {
+      this.currentBindings = YAML.parse(element.textContent);
+    }
+
+    this.currentResult = undefined;
+    if (this.currentProgram) {
+      this.error = undefined;
+
+      const result = this.currentProgram(this.currentBindings);
+
+      if (CEL.isCelError(result)) {
+        throw result;
+      }
+
+      this.currentResult = result;
+
+      const jsonCompatibleResult = toJsonCompatibleValue(result);
+      this.currentResultYAML = YAML.stringify(jsonCompatibleResult);
+    }
+  }
+}
+
+function toJsonCompatibleValue(
+  value: CEL.CelValue,
+): ReturnType<(typeof JSON)["parse"]> {
+  if (CEL.isCelMap(value)) {
+    return Object.fromEntries(
+      [...value.entries()].map(
+        ([k, v]): [string, ReturnType<(typeof JSON)["parse"]>] => [
+          k.toString(),
+          toJsonCompatibleValue(v),
+        ],
+      ),
+    );
+  }
+
+  if (CEL.isCelList(value)) {
+    return [...value].map(
+      (v): ReturnType<(typeof JSON)["parse"]> => toJsonCompatibleValue(v),
+    );
+  }
+
+  return value;
```

Comment:
I think we also need to handle CelUint, bigint, Uint8Array, maybe also CelType and ReflectMessage: https://github.com/bufbuild/cel-es/tree/main/packages/cel#types

This could be a replacer: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#the_replacer_parameter
YAML stringify supports it too.

Or it could be a function in @bufbuild/cel - but then we need to be very certain about it.

Or we could just have a comment here to clarify that this function doesn't handle all possible CEL values.